server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -168,6 +168,9 @@ protected boolean processField(RootObjectMapper.Builder builder, String fieldNam
                       }
                   ]
                 */
+                if ((fieldNode instanceof List) == false) {
+                    throw new MapperParsingException("Dynamic template syntax error, expects an array of named objects");
```

Comment:
```suggestion
                    throw new MapperParsingException("Dynamic template syntax error. An array of named objects is expected.");
```
small wording nit, with "expects" in the previous example its not clear what the subject is (i.e. what expects the array. Using passive voice this should be clearer.